qiskit/_result.py

```diff
@@ -33,7 +33,6 @@ class Result(object):
 
     Internal::
 
-        qobj =  { -- the quantum object that was complied --}
         result = {
```

Comment:
This sounds ok to me.

Reply:
Maybe we can leave this for a subsequent PR since it's not directly related to this one?

Reply:
yeah i think we should leave this, also because it's not clear how qobj and result may be represented together later.
I think the most important thing in this PR is to not give errors when `metadata` is unavailable from a past job.